core/src/main/java/com/linecorp/armeria/common/multipart/StreamMessages.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.multipart;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import org.reactivestreams.Publisher;
+
+import com.linecorp.armeria.common.stream.PublisherBasedStreamMessage;
+import com.linecorp.armeria.common.stream.StreamMessage;
+
+final class StreamMessages {
```

Comment:
Should we move the methods in this class to `StreamMessage`? Like we have `HttpResponse.of(Publisher<? extends HttpObject>)`, we could have:

- `StreamMessage.of(Publisher<...>)`
- `StreamMessage.of(Publisher<...>...)`
- `StreamMessage.of(StreamMessage<...>)`
- `StreamMessage.of(StreamMessage<...>...)`

And maybe in a separate PR:
- `HttpResponse.of(StreamMessage<? extends HttpObject>)` 
- `HttpResponse.of(StreamMessage<? extends HttpObject>...)`